katharsis-core/src/main/java/io/katharsis/queryspec/DefaultQuerySpecDeserializer.java

```diff
@@ -240,9 +240,13 @@ private void deserializeFilter(QuerySpec querySpec, Parameter parameter) {
 		Class<?> attributeType = getAttributeType(querySpec, attributePath);
 		Set<Object> typedValues = new HashSet<>();
 		for (String stringValue : parameter.values) {
-			@SuppressWarnings({ "unchecked", "rawtypes" })
-			Object value = typeParser.parse(stringValue, (Class) attributeType);
-			typedValues.add(value);
+			if (stringValue.toLowerCase().equals("null")) {
+				typedValues.add(null);
+			} else {
+				@SuppressWarnings({ "unchecked", "rawtypes" })
+				Object value = typeParser.parse(stringValue, (Class) attributeType);
+				typedValues.add(value);
```

Comment:
Yes, you are right - It's a bit confusing doing code review without full context